packages/ember-core/src/components/form/search.gts

```diff
@@ -0,0 +1,428 @@
+    if (this.activeIndex == -1) {
+      return;
+    }
+    const childElements = Array.from(
+      this.menuElement?.querySelectorAll(`li`) ?? [],
+    );
+    const activeElement = childElements[this.activeIndex];
+    if (!activeElement) {
+      return;
+    }
+    activeElement.scrollIntoView({ block: 'nearest' });
+  }
+
+  onQuery = restartableTask(async (searchString) => {
+    await timeout(this.searchTimeout);
+    this.options = await this.args.query(searchString);
+    this.isFocused = true;
+  });
+
+  @action
+  selectOption(option: SearchOption<T>, index: number, evt?: Event) {
+    evt?.preventDefault();
+    evt?.stopPropagation();
+
+    this.activeIndex = index;
+    this.selectedOption = option;
+
+    this.onBlur();
+  }
+
+  @action
+  moveUp(evt: KeyboardEvent) {
+    evt.preventDefault();
+    evt.stopPropagation();
+
+    if (this.activeIndex > 0) {
+      this.activeIndex--;
+      this.scrollActiveOptionIntoView();
+    }
+  }
+
+  @action
+  moveDown(evt: KeyboardEvent) {
+    evt.preventDefault();
+    evt.stopPropagation();
+
+    if (this.activeIndex < this.options.length - 1) {
+      this.activeIndex++;
+      this.scrollActiveOptionIntoView();
+    }
+  }
+
+  @action
+  enterKeyHandler(evt?: KeyboardEvent) {
+    evt?.preventDefault();
+    evt?.stopPropagation();
+
+    if (this.activeIndex == -1) {
+      return;
+    }
+
+    const option = this.internalOptions[this.activeIndex];
+
+    if (option != undefined) {
+      this.selectOption(option, this.activeIndex);
+    }
+
+    this.onBlur();
+  }
+
+  @action
+  exitKeyHandler(evt?: KeyboardEvent) {
+    evt?.preventDefault();
+    evt?.stopPropagation();
+
+    this.onBlur();
+  }
+
+  @action
+  onFocus(evt: FocusEvent) {
+    evt.preventDefault();
+    evt.stopPropagation();
+
+    this.isFocused = true;
+  }
+
+  @action
+  onBlur() {
+    this.isFocused = false;
+    this.searchInputElement?.blur();
+  }
+
+  @action
+  onSearch(evt: Event) {
+    this.searchString = (evt.target as HTMLInputElement).value;
+    this.value = '';
+    this.activeIndex = -1;
+
+    if (!this.canPerformSearch) {
+      return;
+    }
+
+    this.onQuery.perform(this.searchString);
+  }
+
+  @action
+  clear() {
+    this.searchString = '';
+    this.value = '';
+    this.activeIndex = -1;
+
+    this.onBlur();
+  }
+
+  @action
+  onSearchBarInsert(element: HTMLElement) {
+    this.searchInputElement = element;
+  }
+
+  @action
+  onMenuInsert(element: HTMLElement) {
+    this.menuElement = element;
+  }
+
+  <template>
+    <div class={{this.classList}} {{onClickOutside this.onBlur}} ...attributes>
+      <div class="input-group">
+        {{#unless this.hideSearchIcon}}
+          <span class="input-group-text">
+            {{#if this.loading}}
+              <span class="spinner-border spinner-border-sm" />
+            {{else}}
+              <i class="bi bi-search" />
+            {{/if}}
+          </span>
+        {{/unless}}
+        <input
```

Comment:
Is there any reason this doesn't use the `TextInput` component, since it already does a lot of this for you?